src/components/hw-design/hw-icon-button.tsx

```diff
@@ -0,0 +1,44 @@
+import { IconNode, LucideIcon } from "lucide-react";
+import { ReactNode } from "react";
+
+/**
+ * A reusable Hack Western button component with variants and customizable styles and attributes.
+ * To override any of the default tailwind styles, pass in custom styles with the className prop, use ! before a className to force style override
+ * @param props - The props for the Button component.
+ * @param props.className - Additional CSS classes to apply to the button.
+ */
+function HWIconButton({
+  text,
+  className = "",
+  variant,
+  icon: Icon,
```

Comment:
I think this component is unnecessary. whenever we need an icon, we can already include that as a child of the button element (or create a custom reusable button component with that icon inside.

Reply:
just wanna avoid abstraction hell ya know

Reply:
Ok, from your previous comment I'm fine with this component then